examples/02_decoding/plot_haxby_understand_decoder.py

```diff
@@ -0,0 +1,330 @@
+
+from sklearn.metrics import roc_auc_score
+from sklearn.model_selection import LeaveOneGroupOut
+
+logo_cv = LeaveOneGroupOut()
+
+# Transform fMRI data into a 2D numpy array and standardize it with the masker
+X = masker.fit_transform(fmri_img)
+print(f"fMRI data shape after masking: {X.shape}")
+# So now we have a 2D numpy array of shape (864, 464) where each row
+# corresponds to a trial and each column corresponds to a feature
+# (voxel in the Ventral Temporal cortex).
+
+# Loop over each CV split and each class vs. rest binary classification
+# problems (number of classification problems = n_labels)
+scores_sklearn = []
+for klass in range(n_labels):
+    for train, test in logo_cv.split(X, y, groups=run):
+        # separate train and test events in the data
+        X_train, X_test = X[train], X[test]
+        # separate labels for train and test events for a given class vs. rest
+        # problem
+        y_train, y_test = y_binary[train, klass], y_binary[test, klass]
+
+        # select the voxels by fitting feature selector on training data
+        X_train = feature_selector.fit_transform(X_train, y_train)
+        # pick the same voxels in the test data
+        X_test = feature_selector.transform(X_test)
+
+        # fit the classifier on the training data
+        classifier.fit(X_train, y_train)
+        # predict the labels on the test data
+        pred = classifier.predict_proba(X_test)
+
+        # calculate the ROC AUC score
+        score = roc_auc_score(y_test, pred[:, 1])
+        scores_sklearn.append(score)
+
+# %%
+# Decode via the :class:`nilearn.decoding.Decoder`
+# ------------------------------------------------
+#
+# All these steps can be done in a few lines and made faster via parallel
+# processing using the ``n_jobs`` parameter in
+# :class:`nilearn.decoding.Decoder`.
+
+from nilearn.decoding import Decoder
+
+decoder = Decoder(
+    estimator="logistic_l2",
+    mask=mask_vt,
+    standardize="zscore_sample",
+    n_jobs=n_labels,
+    cv=logo_cv,
+    screening_percentile=20,
+    scoring="roc_auc_ovr",
+)
+decoder.fit(fmri_img, y, groups=run)
+scores_nilearn = np.concatenate(list(decoder.cv_scores_.values()))
+
+# %%
+# Compare the results
+# -------------------
+#
+# Let's compare the results from the Scikit-Learn pipeline and the Nilearn
+# decoder.
+
+print("Nilearn mean AU-ROC score", np.mean(scores_nilearn))
+print("Scikit-Learn mean AU-ROC score", np.mean(scores_sklearn))
+
+# %%
+# As we can see, the mean AU-ROC scores from the Scikit-Learn pipeline and
+# Nilearn's :class:`nilearn.decoding.Decoder` are identical.
+#
+# The advantage of using Nilearn's :class:`nilearn.decoding.Decoder` is
+# that it does all these steps under the hood and provides a simple interface
+# to train, cross-validate and predict on new data, while also parallelizing
+# the computations to make the cross-validation faster. It also organises the
+# results in a structured way that can be easily accessed and analysed.
```

Comment:
maybe link to this new example from this file: examples/00_tutorials/plot_decoding_tutorial.py

This way when people are done with the basic intro they can look deeper with this new example.